### PR TITLE
Scale test script improvements.

### PR DESCRIPTION
**Pull request by mpereira**
- [X] Black formatting.
- [X] Show aggregated test information.
- [X] List possibly existing artifacts for the test before running script.
- [X] Rebuild docker image in case of partially set up image.
- [X] Put script run logs under `scale-tests/$test_name/script_logs` instead of repo root.

These changes were tested during the MWT #9.

### CODE REVIEW

File: scale-tests/run.sh [hunk 1]
```diff
@@ -132,12 +209,25 @@ if is_interactive; then
...
+readonly container_finished_setting_up=$?
+set -e
 
-  docker build -t "${IMAGE_NAME}" "${TEST_DIRECTORY}/scale-tests" | tee -a "${LOG_FILE}"
+if [ ${container_running} -ne 0 ] || [ ${container_finished_setting_up} -ne 0 ]; then
+  log "Building Docker image for ${TEST_NAME}"
+
+  log "Cleaning up possibly pre-existing containers"
+  docker kill "${CONTAINER_NAME}" > /dev/null 2>&1 || true
+  docker rm -f "${CONTAINER_NAME}" > /dev/null 2>&1 || true
```
Comment (samvantran) [inline scale-tests/run.sh]: Nice, I had been doing this myself and telling others to do it too whenever issues popped up in the script.

File: scale-tests/run.sh [hunk 2]
```diff
@@ -113,6 +120,76 @@ function container_exec () {
...
+readonly NON_GPU_TOTAL_QUOTA_DRIVERS_MEM=$((NON_GPU_NUM_DISPATCHERS * NON_GPU_QUOTA_DRIVERS_MEM))
+readonly NON_GPU_TOTAL_QUOTA_EXECUTORS_CPUS=$((NON_GPU_NUM_DISPATCHERS * NON_GPU_QUOTA_EXECUTORS_CPUS))
+readonly NON_GPU_TOTAL_QUOTA_EXECUTORS_MEM=$((NON_GPU_NUM_DISPATCHERS * NON_GPU_QUOTA_EXECUTORS_MEM))
+
+readonly GPU_TOTAL_QUOTA_DRIVERS_CPUS=$((GPU_NUM_DISPATCHERS * GPU_QUOTA_DRIVERS_CPUS))
+readonly GPU_TOTAL_QUOTA_DRIVERS_MEM=$((GPU_NUM_DISPATCHERS * GPU_QUOTA_DRIVERS_MEM))
+readonly GPU_TOTAL_QUOTA_DRIVERS_GPUS=$((GPU_NUM_DISPATCHERS * GPU_QUOTA_DRIVERS_GPUS))
+readonly GPU_TOTAL_QUOTA_EXECUTORS_CPUS=$((GPU_NUM_DISPATCHERS * GPU_QUOTA_EXECUTORS_CPUS))
+readonly GPU_TOTAL_QUOTA_EXECUTORS_MEM=$((GPU_NUM_DISPATCHERS * GPU_QUOTA_EXECUTORS_MEM))
+readonly GPU_TOTAL_QUOTA_EXECUTORS_GPUS=$((GPU_NUM_DISPATCHERS * GPU_QUOTA_EXECUTORS_GPUS))
```
Comment (samvantran) [inline scale-tests/run.sh]: AFAIK we still remove GPU executor quota for each test. Do we need L139-141? Then again, I guess it's good to have for future cases where we might specify executor quota again.

Comment (mpereira) [reply scale-tests/run.sh]: Right. Even thought we're not creating GPU executor quota it is useful to have the script confirming that it is the case.

File: scale-tests/deploy-batch-marathon-app.py [hunk 3]
```diff
@@ -51,9 +69,7 @@ def _get_app_defn(app_id: str, dcos_username: str, dcos_password: str, input_fil
             "cmd": "cd $MESOS_SANDBOX; git clone https://github.com/mesosphere/spark-build.git; cd spark-build/; pwd; git checkout $SPARK_BUILD_BRANCH; python3 -m venv test-env; pip3 install -r scale-tests/requirements.txt; dcos cluster setup https://master.mesos --username=$DCOS_UID --password=$DCOS_PASSWORD --no-check; dcos package install spark --cli --yes; cd scale-tests/; export PYTHONPATH=../spark-testing:../testing; python3 batch_test.py $MESOS_SANDBOX/$SCRIPT_ARGS",
             "container": {
                 "type": "DOCKER",
-                "docker": {
-                    "image": "susanxhuynh/dcos-commons:spark",
-                }
+                "docker": {"image": "susanxhuynh/dcos-commons:spark"},
```
[the remark this answers is not in the document]
Comment (mpereira) [reply scale-tests/deploy-batch-marathon-app.py]: Good point. I've created [DCOS-43786 ](https://jira.mesosphere.com/browse/DCOS-43786) for this.

**Pull request merged**